robotiq_force_torque_sensor/nodes/rq_sensor.cpp

```diff
@@ -40,99 +41,233 @@
+        _wrench_msg.wrench.force.y = msgStream.Fy;
+        _wrench_msg.wrench.force.z = msgStream.Fz;
+        _wrench_msg.wrench.torque.x = msgStream.Mx;
+        _wrench_msg.wrench.torque.y = msgStream.My;
+        _wrench_msg.wrench.torque.z = msgStream.Mz;
+        _wrench_pub.publish(_wrench_msg);
+    }
+}
+
+bool SensorBase::_receiverCallback(robotiq_force_torque_sensor::sensor_accessor::Request &req,
+                                   robotiq_force_torque_sensor::sensor_accessor::Response &res)
+{
+    ROS_INFO("I heard: [%s]", req.command.c_str());
+    INT_8 buffer[512];
+    _decodeMessageAndDo((char *)req.command.c_str(), buffer);
+    res.res = buffer;
+    ROS_INFO("I send: [%s]", res.res.c_str());
+    return true;
+}
 
 /**
  * \brief Decode the message received and do the associated action
  * \param buff message to decode
  * \param ret buffer containing the return value from a GET command
  */
-static void decode_message_and_do(INT_8 const  * const buff, INT_8 * const ret)
+void RealSensor::_decodeMessageAndDo(INT_8 const *const buff, INT_8 *const ret)
 {
     INT_8 get_or_set[3];
     INT_8 nom_var[4];
 
-    if(buff == NULL || strlen(buff) != 7)
-    {
+    if (buff == NULL || strlen(buff) != 7) {
         return;
     }
 
     strncpy(get_or_set, &buff[0], 3);
-    strncpy(nom_var, &buff[4], strlen(buff) -3);
+    strncpy(nom_var, &buff[4], strlen(buff) - 3);
 
-    if(strstr(get_or_set, "GET"))
-    {
+    if (strstr(get_or_set, "GET")) {
         rq_state_get_command(nom_var, ret);
     }
-    else if(strstr(get_or_set, "SET"))
-    {
-        if(strstr(nom_var, "ZRO"))
-        {
+    else if (strstr(get_or_set, "SET")) {
+        if (strstr(nom_var, "ZRO")) {
             rq_state_do_zero_force_flag();
-            strcpy(ret,"Done");
+            strcpy(ret, "Done");
         }
     }
 }
 
-bool receiverCallback(robotiq_force_torque_sensor::sensor_accessor::Request& req,
-    robotiq_force_torque_sensor::sensor_accessor::Response& res)
-{
-    ROS_INFO("I heard: [%s]",req.command.c_str());
-    INT_8 buffer[512];
-    decode_message_and_do((char*)req.command.c_str(), buffer);
-    res.res = buffer;
-    ROS_INFO("I send: [%s]", res.res.c_str());
-    return true;
-}
-
 /**
- * \fn bool wait_for_other_connection()
+ * \fn bool waitForOtherConnection()
  * \brief Each second, checks for a sensor that has been connected
  */
-bool wait_for_other_connection(void)
+bool RealSensor::_waitForOtherConnection(void)
 {
-    INT_8 ret;
-    INT_8 count = 0;
+    uint8_t tries_left = 6;
 
-    while(ros::ok() && count < 6)
-    {
-        if (count % 2 == 0)
-        {
+    while (ros::ok() && tries_left > 0) {
+        if (tries_left-- % 2 == 0) {
             ROS_INFO("Waiting for sensor connection...");
         }
-        usleep(10000); //Attend 1 seconde.
-
-        ret = rq_sensor_state(max_retries_);
-        if(ret == 0)
-        {
+        if (rq_sensor_state(_max_retries) == 0) {
```

Comment:
You removed the 0.01s sleep here. For now, I would leave it in

Reply:
Sleep was moved such that it would be done only if the first attempt failed. Rather than before every attempt, if first attempt is likely to fail w/o it previous order can be restored.